tests/test_native_mcm.py

```diff
@@ -230,181 +175,162 @@ def obs_tape(x, y, z, reset=False, postselect=None):
-
-    dev = qml.device(device_name, wires=2, shots=shots)
-    dq = qml.device("default.qubit", shots=shots)
-    params = [np.pi / 7, np.pi / 6, -np.pi / 5]
-    obs = qml.PauliZ(0) @ qml.PauliY(1)
-
-    def func(x, y, z):
-        mcms = obs_tape(x, y, z, reset=reset, postselect=postselect)
-        qml.cond(mcms[0] != mcms[1], qml.RY)(z, wires=0)
-        qml.cond(mcms[0] == mcms[1], qml.RY)(z, wires=1)
+        obs = (
+            (m0 - 2 * m1) * m2 + 7
+            if mcm_f == "mix"
+            else ([m0, m1, m2] if mcm_f == "list" else mcm_f(m2))
+        )
         return measure_f(op=obs)
 
-    func1 = qml.QNode(func, dev)
-    func2 = qml.defer_measurements(qml.QNode(func, dq))
-
-    results1 = func1(*params)
-    results2 = func2(*params)
+    results1 = qml.QNode(func, dev, mcm_method="one-shot")(param)
+    results2 = qml.QNode(func, dq, mcm_method="deferred")(param)
 
-    if postselect is None or measure_f in (qml.expval, qml.probs, qml.var):
-        validate_measurements(measure_f, shots, results1, results2)
+    validate_measurements(measure_f, shots, results1, results2)
 
 
-@flaky(max_runs=5)
-@pytest.mark.parametrize("shots", [5000, [5000, 5001]])
-@pytest.mark.parametrize("postselect", [None, 0, 1])
-@pytest.mark.parametrize("reset", [False, True])
-@pytest.mark.parametrize("measure_f", [qml.counts, qml.probs, qml.sample])
-def test_composite_mcm_measure_value_list(shots, postselect, reset, measure_f):
-    """Tests that LightningDevice handles a circuit with a composite mid-circuit measurement and a
-    conditional gate. A single measurement of a composite mid-circuit measurement is performed
-    at the end."""
+@pytest.mark.parametrize(
+    "mcm_f",
+    [
+        lambda x, y: x + y,
+        lambda x, y: x - 7 * y,
+        lambda x, y: x & y,
+        lambda x, y: x == y,
+        lambda x, y: 4.0 * x + 2.0 * y,
+    ],
+)
+def test_counts_return_type(mcm_f):
+    """Tests that LightningQubit returns the same keys for ``qml.counts`` measurements with ``dynamic_one_shot`` and ``defer_measurements``."""
+    shots = 500
 
-    dev = qml.device(device_name, wires=2, shots=shots)
     dq = qml.device("default.qubit", shots=shots)
+    dev = get_device(wires=3, shots=shots)
     param = np.pi / 3
 
+    @qml.qnode(dev)
     def func(x):
         qml.RX(x, 0)
         m0 = qml.measure(0)
         qml.RX(0.5 * x, 1)
-        m1 = qml.measure(1, reset=reset, postselect=postselect)
+        m1 = qml.measure(1)
         qml.cond((m0 + m1) == 2, qml.RY)(2.0 * x, 0)
-        m2 = qml.measure(0)
-        return measure_f(op=[m0, m1, m2])
+        return qml.counts(op=mcm_f(m0, m1))
 
-    func1 = qml.QNode(func, dev)
-    func2 = qml.defer_measurements(qml.QNode(func, dq))
-
-    results1 = func1(param)
-    results2 = func2(param)
-
-    validate_measurements(measure_f, shots, results1, results2)
+    results1 = qml.QNode(func, dev, mcm_method="one-shot")(param)
```

Comment:
Can we pass `diff_method` to `qml.QNode` as well?

Reply:
Should not be an issue no, but not all diff methods are supported. Even parameter-shift isn't supported with post-selection not `None`. Anything in mind?

Reply:
Nothing specifically. Just want to understand more about this feature